api/provider/aws/deploy_create.go

```diff
@@ -47,6 +47,24 @@ func (d *DeployProvider) createTaskDefinition(taskDefinition *ecs.TaskDefinition
 		input.SetNetworkMode(nm)
 	}
 
+	// Because of the logic in renderTaskDefinition(), taskDefinition.RequiresCompatibilities
+	// should always contain one or both of "EC2" and "FARGATE"
+	input.SetRequiresCompatibilities(taskDefinition.RequiresCompatibilities)
+
+	if cpu := aws.StringValue(taskDefinition.Cpu); cpu != "" {
+		input.SetCpu(cpu)
+	}
+
+	if mem := aws.StringValue(taskDefinition.Memory); mem != "" {
+		input.SetMemory(mem)
+	}
+
+	// Hard-coding an ARN isn't usually the best thing, but we know exactly what this ARN will look like
+	// and the only variable in it is the account ID, which Layer0 already knows about.
+	accountID := d.Config.AccountID()
```

Comment:
It's created as part of `l0-setup`

Reply:
To clarify my answer a bit more, this execution role is something we were already creating as part of `l0-setup`; it's needed elsewhere by ECS. You can see this role being explicitly called out in https://github.com/quintilesims/layer0/pull/567/commits/a73f23457940ce0ede50c1128d9b4a46252217c7 for instance. Why does it need to be called out here? It's a Fargate launch type requirement.

Reply:
@sesh-kebab It's set here because it's a requirement to execute Fargate services and tasks. Maybe it ought to be moved into a fargate-conditional? It hasn't hurt anything by being out in the open, though. :p

Now that you bring it up, though, I don't _know_ if we can guarantee that this role exists. The docs say that it's created as part of the console first-run walkthrough, but that's probably not guaranteed to have occurred for every user. I think @diemonster made the discovery (and implementation) about needing this role for Fargate; maybe he has additional insight about whether or not we can guarantee it, and if not, what the best solution to making that guarantee would be.